app/ui/legacy/src/main/java/com/fsck/k9/activity/MessageList.kt

```diff
@@ -1406,9 +1403,7 @@ open class MessageList :
         this.search = search
         singleFolderMode = false
 
-        if (search!!.searchAllAccounts()) {
-            account = null
-        } else {
+        if (! search!!.searchAllAccounts()) {
```

Comment:
Nit: I'm surprised ktlint doesn't flag this. There shouldn't be a space between `!` and `search…`.